packages/graphql/src/translate/queryAST/factory/OperationFactory.ts

```diff
@@ -925,16 +925,74 @@ export class OperationsFactory {
         return updateOp;
     }
 
-    private createCustomCypherOperation({
+    public createCustomCypherOperation({
+        resolveTree,
+        context,
+        entity,
+        cypherAttributeField,
+        cypherArguments = {},
+    }: {
+        resolveTree: ResolveTree;
+        context: Neo4jGraphQLTranslationContext;
+        entity?: EntityAdapter;
+        cypherAttributeField: AttributeAdapter;
+        cypherArguments?: Record<string, any>;
+    }): CypherOperation | CompositeCypherOperation | CypherScalarOperation {
+        if (!entity) {
+            const selection = new CustomCypherSelection({
+                operationField: cypherAttributeField,
+
+                rawArguments: cypherArguments,
+                isNested: true,
+            });
+            return new CypherScalarOperation(selection, cypherAttributeField, true);
+        }
+        if (isConcreteEntity(entity)) {
+            const selection = new CustomCypherSelection({
+                operationField: cypherAttributeField,
+
+                rawArguments: cypherArguments,
+                isNested: true,
+            });
+            const customCypher = new CypherOperation({
+                cypherAttributeField: cypherAttributeField,
+                target: entity,
+                selection,
+            });
+            return this.hydrateReadOperation({ entity, operation: customCypher, resolveTree, context, whereArgs: {} });
+        }
+        const selection = new CustomCypherSelection({
+            operationField: cypherAttributeField,
+            rawArguments: cypherArguments,
+            isNested: true,
+        });
+
+        const CypherReadPartials = entity.concreteEntities.map((concreteEntity) => {
+            const partialSelection = new NodeSelection({ target: concreteEntity, useContextTarget: true });
+            const partial = new CompositeReadPartial({ target: concreteEntity, selection: partialSelection });
+            // The Typename filter here is required to access concrete entities from a Cypher Union selection.
+            // It would be probably more ergonomic to pass the label filter with the selection,
+            // although is currently not possible to do so with Cypher.Builder
```

Comment:
should this be a feature request on cypher builder?

Reply:
It is!